### DISCUSSION TITLE
Enable Alerting in Superset

### DISCUSSION THREAD

**Discussion by codegeekjk** [apache/superset#13127] [General]
I am trying to enable Alerting in Superset I added the following config.
```
RESULTS_BACKEND = RedisCache(host=REDIS_HOST, port=REDIS_PORT, key_prefix='superset_results')

EMAIL_REPORTS_WEBDRIVER = 'firefox'
EMAIL_REPORTS_USER = 'Admin'
WEBDRIVER_BASEURL = "http://0.0.0.0:8088/"

ENABLE_ALERTS = True

THUMBNAIL_SELENIUM_USER = "Admin"

DATA_CACHE_CONFIG = {
    'CACHE_TYPE': 'redis',
    'CACHE_DEFAULT_TIMEOUT': 60 * 60 * 24, # 1 day default (in secs)
    'CACHE_REDIS_URL': f'redis://{REDIS_HOST}:{REDIS_PORT}/0',
}

class CeleryConfig(object):
    BROKER_URL = 'redis://{REDIS_HOST}:{REDIS_PORT}/0'
    CELERY_IMPORTS = (
        'superset.sql_lab',
        'superset.tasks',
    )
    CELERY_RESULT_BACKEND = 'redis://{REDIS_HOST}:{REDIS_PORT}/0'
    CELERYD_LOG_LEVEL = 'DEBUG'
    CELERYD_PREFETCH_MULTIPLIER = 10
    CELERY_ACKS_LATE = True
    CELERY_ANNOTATIONS = {
        'sql_lab.get_sql_results': {
            'rate_limit': '100/s',
        },
        'email_reports.send': {
            'rate_limit': '1/s',
            'time_limit': 120,
            'soft_time_limit': 150,
            'ignore_result': True,
        },
    }
    CELERYBEAT_SCHEDULE = {
        'email_reports.schedule_hourly': {
            'task': 'email_reports.schedule_hourly',
            'schedule': crontab(minute='*', hour='*'),
        },
        'alerts.schedule_check': {
            'task': 'alerts.schedule_check',
            'schedule': crontab(minute='*', hour='*')
        },
    }

CELERY_CONFIG = CeleryConfig
```

I am getting the following errors:
```
line 57, in CeleryConfig
    'schedule': crontab(minute='*', hour='*'),
NameError: name 'crontab' is not defined
```
I have installed using `pip install python-crontab`.
Any help ?

**Comment by Asturias-sam** [comment]
have you imported the module `from celery.schedules import crontab` ?

**Reply by GangadharGuska** [reply]
> I am trying to enable Alerting in Superset I added the following config.
> 
> ```
> RESULTS_BACKEND = RedisCache(host=REDIS_HOST, port=REDIS_PORT, key_prefix='superset_results')
> 
> EMAIL_REPORTS_WEBDRIVER = 'firefox'
> EMAIL_REPORTS_USER = 'Admin'
> WEBDRIVER_BASEURL = "http://0.0.0.0:8088/"
> 
> ENABLE_ALERTS = True
> 
> THUMBNAIL_SELENIUM_USER = "Admin"
> 
> DATA_CACHE_CONFIG = {
>     'CACHE_TYPE': 'redis',
>     'CACHE_DEFAULT_TIMEOUT': 60 * 60 * 24, # 1 day default (in secs)
>     'CACHE_REDIS_URL': f'redis://{REDIS_HOST}:{REDIS_PORT}/0',
> }
> 
> class CeleryConfig(object):
>     BROKER_URL = 'redis://{REDIS_HOST}:{REDIS_PORT}/0'
>     CELERY_IMPORTS = (
>         'superset.sql_lab',
>         'superset.tasks',
>     )
>     CELERY_RESULT_BACKEND = 'redis://{REDIS_HOST}:{REDIS_PORT}/0'
>     CELERYD_LOG_LEVEL = 'DEBUG'
>     CELERYD_PREFETCH_MULTIPLIER = 10
>     CELERY_ACKS_LATE = True
>     CELERY_ANNOTATIONS = {
>         'sql_lab.get_sql_results': {
>             'rate_limit': '100/s',
>         },
>         'email_reports.send': {
>             'rate_limit': '1/s',
>             'time_limit': 120,
>             'soft_time_limit': 150,
>             'ignore_result': True,
>         },
>     }
>     CELERYBEAT_SCHEDULE = {
>         'email_reports.schedule_hourly': {
>             'task': 'email_reports.schedule_hourly',
>             'schedule': crontab(minute='*', hour='*'),
>         },
>         'alerts.schedule_check': {
>             'task': 'alerts.schedule_check',
>             'schedule': crontab(minute='*', hour='*')
>         },
>     }
> 
> CELERY_CONFIG = CeleryConfig
> ```
> 
> I am getting the following errors:
> 
> ```
> line 57, in CeleryConfig
>     'schedule': crontab(minute='*', hour='*'),
> NameError: name 'crontab' is not defined
> ```
> 
> I have installed using `pip install python-crontab`.
> Any help ?

I am facing the same issue ,Can anyone help in configuring the Alerts

**Reply by jagamts1** [reply]
@GangadharGuska Just import will works. `from celery.schedules import crontab`

**Comment by carys-cc** [comment]
Where do you put this import statement?

**Reply by armfem** [reply]
At the beginning of the file.
